src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/SyndicationFeedFormatter.cs

```diff
@@ -2,191 +2,169 @@
-
     public delegate bool TryParseDateTimeCallback(XmlDateTimeData data, out DateTimeOffset dateTimeOffset);
     public delegate bool TryParseUriCallback(XmlUriData data, out Uri uri);
 
     [DataContract]
     public abstract class SyndicationFeedFormatter
     {
-        private SyndicationFeed _feed;
-
         protected SyndicationFeedFormatter()
         {
-            _feed = null;
+            Feed = null;
             DateTimeParser = GetDefaultDateTimeParser();
         }
 
         protected SyndicationFeedFormatter(SyndicationFeed feedToWrite)
         {
-            if (feedToWrite == null)
-            {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("feedToWrite");
-            }
-            _feed = feedToWrite;
+            Feed = feedToWrite ?? throw new ArgumentNullException(nameof(feedToWrite));
             DateTimeParser = GetDefaultDateTimeParser();
         }
 
-        public SyndicationFeed Feed
-        {
-            get
-            {
-                return _feed;
-            }
-        }
+        public SyndicationFeed Feed { get; private set; }
 
         public TryParseUriCallback UriParser { get; set; } = DefaultUriParser;
 
         // Different DateTimeParsers are needed for Atom and Rss so can't set inline
         public TryParseDateTimeCallback DateTimeParser { get; set; }
 
-        internal virtual TryParseDateTimeCallback GetDefaultDateTimeParser()
-        {
-            return NotImplementedDateTimeParser;
-        }
+        internal virtual TryParseDateTimeCallback GetDefaultDateTimeParser() => NotImplementedDateTimeParser;
 
         private bool NotImplementedDateTimeParser(XmlDateTimeData XmlDateTimeData, out DateTimeOffset dateTimeOffset)
         {
-            dateTimeOffset = default(DateTimeOffset);
+            dateTimeOffset = default;
             return false;
         }
 
-        public abstract string Version
-        { get; }
+        public abstract string Version { get; }
 
         public abstract bool CanRead(XmlReader reader);
 
         public abstract void ReadFrom(XmlReader reader);
 
         public override string ToString()
         {
-            return String.Format(CultureInfo.CurrentCulture, "{0}, SyndicationVersion={1}", this.GetType(), this.Version);
+            return string.Format(CultureInfo.CurrentCulture, "{0}, SyndicationVersion={1}", GetType(), Version);
         }
 
         public abstract void WriteTo(XmlWriter writer);
 
         internal static protected SyndicationCategory CreateCategory(SyndicationFeed feed)
         {
-            if (feed == null)
-            {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("feed");
-            }
-            return GetNonNullValue<SyndicationCategory>(feed.CreateCategory(), SR.FeedCreatedNullCategory);
+            Debug.Assert(feed != null);
+            return GetNonNullValue(feed.CreateCategory(), SR.FeedCreatedNullCategory);
```

Comment:
There is a code path where feed could be null so this should remain a release build runtime check and not a debug build assertion.